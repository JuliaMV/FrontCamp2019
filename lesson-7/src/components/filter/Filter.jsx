import React from 'react';
import PropTypes from 'prop-types';

import css from './Filter.module.css';

const Filter = ({ title, fields }) => (
  <div className={css.filters}>
    <span className={css.title}>{ title }</span>
    <ul className={css.filtersList}>
      {fields.map((field) => (
        <li key={field} className={css.filtersItem}>
          <button type="button" className={css.filtersButton}>{field}</button>
        </li>
      ))}
    </ul>
  </div>
);

Filter.propTypes = {
  title: PropTypes.string.isRequired,
  fields: PropTypes.arrayOf(PropTypes.string).isRequired,
  // align: PropTypes.string,
};

// Filter.defaultProps = {
//   align: 'left',
// };


export default Filter;
