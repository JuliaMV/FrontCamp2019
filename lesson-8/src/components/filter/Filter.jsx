import React from 'react';
import PropTypes from 'prop-types';

import css from './Filter.module.css';

const getFields = (fields, activeFilter, filterHandler) => {
  console.log(fields, activeFilter, filterHandler);
  return fields.map((field) => (
    <li key={field} className={css.filtersItem}>
      <button type="button" className={field === activeFilter ? css.filtersButtonActive : css.filtersButton} onClick={() => filterHandler(field)}>{field}</button>
    </li>
  ));
};

const Filter = (props) => {
  const {
    title, fields, activeFilter, filterHandler,
  } = props;
  console.log(props);
  const fildsList = getFields(fields, activeFilter, filterHandler);
  console.log(fildsList);
  return (
    <div className={css.filters}>
      <span className={css.title}>{ title }</span>
      <ul className={css.filtersList}>
        {fildsList}
      </ul>
    </div>
  );
};

Filter.propTypes = {
  title: PropTypes.string.isRequired,
  fields: PropTypes.arrayOf(PropTypes.string).isRequired,
  activeFilter: PropTypes.string.isRequired,
  filterHandler: PropTypes.func.isRequired,
};

export default Filter;
