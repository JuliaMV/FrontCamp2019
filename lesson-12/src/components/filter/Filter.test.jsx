import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Filter from './Filter';

Enzyme.configure({ adapter: new Adapter() });

describe('Filter component', () => {
  test('should render with default props', () => {
    const component = shallow(<Filter />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
