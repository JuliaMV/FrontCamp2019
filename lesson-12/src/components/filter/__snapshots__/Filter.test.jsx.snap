// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should render with default props 1`] = `
<div>
  <span>
    Search by
  </span>
  <ul>
    <li
      key="title"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        title
      </button>
    </li>
    <li
      key="genre"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        genre
      </button>
    </li>
  </ul>
</div>
`;
